src/intelligence_layer/connectors/data/data.py

```diff
@@ -0,0 +1,245 @@
+    DataError,
+    DataExternalServiceUnavailable,
+    DataForbiddenError,
+    DataInternalError,
+    DataInvalidInput,
+    DataResourceNotFound,
+)
+from intelligence_layer.connectors.data.models import (
+    DataDataset,
+    DataRepository,
+    DataRepositoryCreate,
+    DatasetCreate,
+)
+
+
+class DataClient:
+    """Data Client class to interact with the Data Platform API.
+
+    Attributes:
+        headers: headers used in the request session
+    """
+
+    _status_code_to_exception: ClassVar[dict[int, type[DataError]]] = {
+        HTTPStatus.SERVICE_UNAVAILABLE: DataExternalServiceUnavailable,
+        HTTPStatus.NOT_FOUND: DataResourceNotFound,
+        HTTPStatus.UNPROCESSABLE_ENTITY: DataInvalidInput,
+        HTTPStatus.FORBIDDEN: DataForbiddenError,
+    }
+
+    def __init__(
+        self,
+        token: str | None,
+        base_data_platform_url: str = "http://localhost:8000",
+        session: requests.Session | None = None,
+    ) -> None:
+        """Initialize the Data Client.
+
+        Args:
+            token: Access token
+            base_data_platform_url: Base URL of the Data Platform API
+            session: Requests session.
+        """
+        self._base_data_platform_url = base_data_platform_url
+        self.headers = {
+            **({"Authorization": f"Bearer {token}"} if token is not None else {}),
+        }
+
+        self._session = session or requests.Session()
+        retry_strategy = Retry(
+            total=3,  # Total number of retries
+            backoff_factor=0.5,  # Exponential backoff factor
+            status_forcelist=[
+                HTTPStatus.TOO_MANY_REQUESTS,  # 429,
+                HTTPStatus.INTERNAL_SERVER_ERROR,  # 500,
+                HTTPStatus.BAD_GATEWAY,  # 502,
+                HTTPStatus.SERVICE_UNAVAILABLE,  # 503,
+                HTTPStatus.GATEWAY_TIMEOUT,  # 504,
+            ],  # Retry on these HTTP status codes
+        )
+
+        # Mount the retry strategy to the session
+        adapter = HTTPAdapter(max_retries=retry_strategy)
+        self._session.mount("http://", adapter)
+        self._session.mount("https://", adapter)
+
+        warnings.warn("DataClient still in beta version and subject to change")
+
+    def list_repositories(self, page: int = 0, size: int = 20) -> list[DataRepository]:
+        """List all the repositories.
+
+        Args:
+            page: Page number
+            size: Number of items per page
+
+        Returns:
+            List of DataRepository objects
+        """
+        url = urljoin(self._base_data_platform_url, "api/v1/repositories")
+        query = urlencode({"page": page, "size": size})
+        response = self._do_request("GET", f"{url}?{query}")
+        repositories = response.json()
+        return [
+            DataRepository(**repository) for repository in repositories["repositories"]
+        ]
+
+    def create_repository(self, repository: DataRepositoryCreate) -> DataRepository:
```

Comment:
what is your opinion on the `DataRepositoryCreate` object? I personally don't use data objects all that much and have not seen them in python code, so I would maybe argue against it

Reply:
FastAPI has a nice page about that; it happens when you have different models for reading(GET) and writing(POST/PUT). For example: https://fastapi.tiangolo.com/tutorial/extra-models/

Reply:
what i meant is we make it the user's job to create an extra object which is passed into the method instead of passing in all the attributes that make up the object and construct it ourselves

Reply:
I try to follow what in the document_index.py connector. For example: https://github.com/Aleph-Alpha/intelligence-layer-sdk/blob/24005b2cc99e6753e295519e494df0c825efe8db/src/intelligence_layer/connectors/document_index/document_index.py#L442

Reply:
ok, works for me